src/lib/Zikula/Core/Theme/AssetBag.php

```diff
@@ -19,18 +39,22 @@ public function __construct(array $assets = array())
      */
     public function add($asset)
     {
+        // ensure value is an array
         if (!is_array($asset)) {
-            $asset = array($asset);
+            $asset = [$asset => self::DEFAULT_WEIGHT];
         }
-        $this->assets = array_merge($this->assets, $asset);
-        $this->assets = array_unique($this->assets);
+        foreach ($asset as $source => $weight) {
+            if ((isset($this->assets[$source]) && $this->assets[$source] > $weight) || !isset($this->assets[$source])) {
+                // keep original weight if lighter. set if not set already.
+                $this->assets[$source] = $weight;
+            }
+        }
+        asort($this->assets); // put array in order by weight
```

Comment:
This is causing trouble, because it doesn't preserve the order of elements with the same weight. I.e. if you have: `['a' => 5, 'b' => 5, 'c' => 5]` and call `asort`, the order could be `['b' => 5, 'a' => 5, 'c' => 5]` afterwards, because the search algorithm is not *stable*. Possible solutions:
- Use a stable sort algorithm: https://github.com/vanderlee/PHP-stable-sort-functions/blob/master/functions/sasort.php#L3
- or don't sort at all but rather insert the new assets at the correct position in line 49.